utils/zerovec/src/varzerovec/owned.rs

```diff
@@ -115,6 +167,118 @@ impl<T: AsVarULE> VarZeroVecOwned<T> {
         &self.entire_slice
     }
 
+    // Invalidate and resize the data at an index, optionally inserting or removing the index.
+    // Also updates affected indices and the length.
+    // Returns a slice to the new element data - it doesn't contain uninitialized data but its value is indeterminate.
+    unsafe fn shift(&mut self, index: usize, new_size: u32, shift_type: ShiftType) -> &mut [u8] {
+        // The format of the encoded data is:
+        //  - four bytes of "len"
+        //  - len*4 bytes for an array of indices
+        //  - the actual data to which the indices point
+        //
+        // When inserting or removing an element, the size of the indices segment must be changed,
+        // so the data before the target element must be shifted by 4 bytes in addition to the
+        // shifting needed for the new element size.
+        let len = self.len();
+        let slice_len = self.entire_slice.len();
+
```

Comment:
issue: if the array is being emptied it should just be set to the empty byte slice (and perhaps add a test to ensure that `.remove()` on a one-element VZV works as expected)

We want VZVs to only ever have a single valid byte representation.

Reply:
Added a test and edge case in remove